frontend/js/src/common/brainzplayer/BrainzPlayer.tsx

```diff
@@ -945,6 +951,75 @@ export default function BrainzPlayer() {
     }
   };
 
+  const handleAlbumMapping = (
+    dataSource: keyof MatchedTrack,
+    releaseMBID: string,
+    album: {
+      trackName: string;
+      uri: string;
+    }[]
+  ): void => {
+    const currentQueue = queueRef.current;
+    const currentAmbientQueue = ambientQueueRef.current;
+
+    // Filter the currentQueue after the current playing track
+    const currentPlayingListenIndex = currentListenIndexRef.current;
+    const currentQueueAfterCurrent = currentQueue.slice(
+      currentPlayingListenIndex + 1
+    );
+
+    const newQueueMatchedTracks: Record<string, string> = {};
+    const newAmbientQueueMatchedTracks: Record<string, string> = {};
+
+    // Filter the tracks in the album that are not yet matched
+    const queueTracksInAlbum = currentQueueAfterCurrent.filter((track) => {
+      return (
+        getReleaseMBID(track) === releaseMBID &&
+        track.matchedTrack?.[dataSource] === undefined
+      );
+    });
+
+    const ambientQueueTracksInAlbum = currentAmbientQueue.filter((track) => {
+      return (
+        getReleaseMBID(track) === releaseMBID &&
+        track.matchedTrack?.[dataSource] === undefined
+      );
+    });
+
+    if (!queueTracksInAlbum.length && !ambientQueueTracksInAlbum.length) {
+      return;
+    }
+
+    const fuzzysearch = new Fuse(album, {
+      keys: ["trackName"],
+    });
+
+    // Find the matches for the tracks in the album
+    queueTracksInAlbum.forEach((track) => {
+      const matches = fuzzysearch.search(getTrackName(track));
+      if (matches[0]) {
+        newQueueMatchedTracks[track.id] = matches[0].item.uri;
+      }
+    });
+
+    ambientQueueTracksInAlbum.forEach((track) => {
+      const matches = fuzzysearch.search(getReleaseName(track));
+      if (matches[0]) {
+        newAmbientQueueMatchedTracks[track.id] = matches[0].item.uri;
+      }
+    });
+
+    // Update the matched tracks
+    dispatch({
+      type: "UPDATE_MATCHED_TRACKS",
+      data: {
+        dataSource,
+        queueMatchedTracks: newQueueMatchedTracks,
```

Comment:
I wonder if it would be easier, instead of storing this info in the listens in the queue, to have a separate object belonging to the BP component, which can be used to retrieve a URI (for a specific music source) using an MBID or name string.
Something like:
```
albumMappings: {
  [releaseMBID || releaseName]: {
    appleMusic: appleMusicAlbumURI,
    spotify: spotifyAlbumURI,
    etc.
  }
}
```
That way, for example if you have the same album spread over two pages of listens, when you go to page 2 you already have the album info you need without having to do the whole process again.
Also means not updating the whole queue, which might re-render the queue UI needlessly.

On the music components side, basically in playListen you would need to call a BP method to check if a value exists in the albumMappings object for this release (by MBID/releasName), and use that if there is a hit; otherwise do the search as usual.

What do you think?

Reply:
There is a problem in this implementation. We'll have a album mapping in this case, which might not be relevant. In order to use a album mapping, we'll have to make an API call to the data source provider every time in order to use this information. 

Right now, what I'm doing is that fetching all the tracks of the current album, and then using the track <> URI mapping to update the upcoming tracks in the queue.

So, if we want to keep this information separate from queueItem, we can save all this information like

```
albumMapping: {
  [releaseName]: {
    [trackName]: {
      appleMusic: appleMusicURI,
      spotify: spotifyURI,
    }
  }
}
```

Reply:
Thank you for the explanation, I misunderstood the code at first.

Do you think my suggestion makes sense,as opposed to modifying the queue items?

Reply:
Yes, this makes sense, as we don't have to re=render the listen cards again.